lib/internal/modules/esm/module_job.js

```diff
@@ -92,7 +97,39 @@ class ModuleJob {
         this.module.instantiate();
       }
     } catch (e) {
-      decorateErrorStack(e);
+      if (StringPrototypeIncludes(e.message,
+                                  ' does not provide an export named')) {
+        const start = StringPrototypeIndexOf(e.message, '\'') + 1;
+        const end =
+          StringPrototypeIndexOf(e.message,
+                                 ' does not provide an export named') - 1;
+        const childSpecifier = StringPrototypeSlice(e.message, start, end);
+        let childFileURL;
+        if (StringPrototypeStartsWith(childSpecifier, '.') ||
+          StringPrototypeStartsWith(childSpecifier, '/')) {
+          // If the specifier is a relative or absolute path make a new
+          // URL object to resolve the file url. This will not work for
+          // bare specifiers.
+          childFileURL = (new URL(childSpecifier, this.module.url)).href;
```

Comment:
This doesn't get the accurate childFileURL... as it is relative to `this.module.url` which is the entry point to the graph, not the specific parents that imported the child. I've shown a bunch of cases where this is "good enough", but with a couple layers of indirection and nested package.json I think it would be possible to inaccurately resolve the format on line 125.

Calling it a night for now but we likely need to make a test for this. The decorated error has the appropriate parent file url... so we may just have to refactor the order of operations and where the error gets decorated to ensure we appropriately resolve the format.